doc/user_guide/model.rst

```diff
@@ -1034,6 +1041,49 @@ on the ``centre`` parameter.
              sigma |  True | 0.00999184 | 2.68064163 |       None |       None
             centre |  True | 9.99980788 | 2.68064070 |        7.0 |       14.0
 
+Multiple linear least squares fitting
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A component is considered linear when its free parameters scale the component only 
+in the y-axis. For the example function ``y = a*x**b``, ``a`` is a linear parameter whilst ``b`` 
+is not. If `b.free = False`, then the component is linear. 
+Components can also be made up of several linear parts. For instance, 
+the 2D-polynomial ``y = a*x**2+b*y**2+c*x+d*y+e`` is entirely linear.
+
+.. note::
+
+    After creating a model with values for the nonlinear parameters, a quick way to set 
+    all nonlinear parameters to be ``free = False`` is to use ``m.set_parameters_not_free(only_nonlinear=True)``
+
+If all components in a model are linear, then a linear optimizer can be used to
+solve the problem as a linear regression problem! This can be done in the standard
+pixel-by-pixel approach used by the other *nonlinear* optimizers in HyperSpy, but
+the main advantage comes in being able to fit the entire dataset in one vectorised
+operation. This can be over a thousand times faster than regular nonlinear fitting, but
+comes with one caveat: All fixed parameters must have the same value across 
+the dataset. That means a gaussian sigma parameter may not have been manually set 
+to vary across the dataset. This is because in order to save on memory, the components
+are calculated once for the current navigation indices and used to fit to all 
+pixels in the navigator.
+
+.. note::
+
+    A good example of a linear model in the electron-microscopy field is an Energy-Dispersive
+    X-ray Spectroscopy (EDS) dataset, which typically consists of a polynomial background and
+    Gaussian peaks with well-defined energy (``Gaussian.centre``) and peak widths 
+    (``Gaussian.sigma``). This dataset can be fit extremely fast with a linear optimizer.
+
+There are three implementations of linear fitting in hyperspy: 
+The 'lstsq' optimizer is the recommended linear optimizer, and supports fitting lazy
```

Comment:
No, it isn't. For line 1073 we can change to "with linear fitting" or " with a linear solver". But I think we should be using the `optimizer=` keyword to distinguish between the linear solvers, rather than introduce e.g. `optimizer='linear'` with an additional `linear_method=` keyword.